Source/Particles/Collision/BinaryCollision.cpp

```diff
@@ -283,12 +287,11 @@ void PairWiseCoulombCollision::doCoulombCollisionsWithinTile
                 auto dV = MathConst::pi*(2.0_rt*ri+1.0_rt)*dr*dr*dz;
 #endif
                 // Call the function in order to perform collisions
-                ElasticCollisionPerez(
+                (*m_binary_collision_functor)(
                     cell_start_1, cell_stop_1, cell_start_2, cell_stop_2,
                     indices_1, indices_2,
                     ux_1, uy_1, uz_1, ux_2, uy_2, uz_2, w_1, w_2,
-                    q1, q2, m1, m2, amrex::Real(-1.0), amrex::Real(-1.0),
-                    dt*ndt, CoulombLog, dV, engine );
+                    q1, q2, m1, m2, dt*ndt, dV, engine );
```

Comment:
For near-future applications, does this need to be generalized, so that it would be able to take as inut other macroparticle quantities (e.g. ionization level, etc.) ?

If so, we could potentially use similar data structures as in our field ionization module, where a general particle data structure is taken as input: 
https://github.com/ECP-WarpX/WarpX/blob/development/Source/Particles/ElementaryProcess/Ionization.H#L89

Reply:
Note that, if the above is indeed needed in the near-term, this is could still be done in a separate PR.

Reply:
Yes that sounds like a very good idea. That can either be done in this PR or in a follow-up, I'm fine with both. (that will most likely be something I need for the fusion module anyways)

Reply:
OK, sounds good. Let us keep this for a future PR.